src/services/datasource.ts

```diff
@@ -204,13 +214,71 @@ class WrappedLokiDatasource extends RuntimeDataSource<DataQuery> {
 
       frames.sort((a, b) => (b.meta?.custom?.sum as number) - (a.meta?.custom?.sum as number));
       subscriber.next({ data: frames, state: LoadingState.Done });
+    } catch (e) {
+      subscriber.next({ data: [], state: LoadingState.Error });
+    }
+
+    return subscriber;
+  }
+
+  private async getDetectedLabels(
+    request: DataQueryRequest<LokiQuery & SceneDataQueryResourceRequest>,
+    ds: DataSourceWithBackend<LokiQuery>,
+    subscriber: Subscriber<DataQueryResponse>
+  ) {
+    const targets = request.targets.filter((target) => {
+      return target.resource === 'detected_labels';
     });
 
+    if (targets.length !== 1) {
+      throw new Error('Detected labels query can only have a single target!');
+    }
+
+    const targetsInterpolated = ds.interpolateVariablesInQueries(targets, request.scopedVars);
+    const interpolatedTarget = targetsInterpolated[0];
```

Comment:
If targets is empty, it could also be empty, so the assertion above that targets.length must be 1, should be fine, until the implementation of `interpolateVariablesInQueries` changes in the loki datasource.

Is it worth adding another assertion that `targetsInterpolated` has length and throwing an error if it doesn't? otherwise we'll likely get an error in the network request

Reply:
Added new error if the result is empty